### PR TITLE
Move iree/compiler to top-level compiler/.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Source files are clean renames (no changes).
* CMake files should be mostly clean renames (just a changed comment about the file path in each).
* Several bazel fixes. Mainly introducing trampoline macros iree_compiler_cc_library, iree_gentbl_cc_library, iree_td_library that inject appropriate includes and deps. And platform_trampoline_deps now takes a repo relative package (vs an //iree relative).
* Uses the CMake "defs" library as done in samples. This is a bit different/better than what was done for runtime/, so I also fixed that one to be the same. So we now have implicit iree::compiler::defs, iree::runtime::defs and iree::samples::defs targets that do the right thing if depended on.
* Re-rigs some absolute paths in the iree-compiler python package build and gets that working in the new layout.
* All deep BUILD files (below iree/compiler/src) were updated with just a combination of sed commands followed by a buildifier fix to clean everything up. I didn't manually tweak any of them (with the exception of the platform_trampoline_deps change mentioned above, in one place).
* Also removes some obsolete "non_darwin" bazelrc thing. I'm not sure why it started yelling at me now (about some bad flag/packages). Folks may need to manually remove those lines from their configured.bazelrc or rerun configure_bazel.py.

### CODE REVIEW

**Comment by jpienaar** [comment]
The review part actually crashes my browser :-/ All of this sounds good, changes mostly looks mechanical though. Couple of clang-format warnings.

I'll try downloading and checking tomorrow morning (mostly just rule macro changes really).

File: iree/build_defs.oss.bzl [hunk 1]
```diff
@@ -46,6 +46,19 @@ def iree_cmake_extra_content(content = "", inline = False):
     """
     pass
 
+def iree_compiler_cc_library(deps = [], **kwargs):
+    """Used for cc_library targets within the //compiler tree.
+
+    This is a pass-through to the native cc_library which adds specific
+    runtime specific options and deps.
```
Comment (jpienaar) [inline iree/build_defs.oss.bzl:53]: compiler specific?

**Review by jpienaar — APPROVED** [review]
LG thanks! (Mobile version worked well and was quite responsive)

**Review by ScottTodd — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/HAL/Target/CUDA/test/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 iree_add_all_subdirs()
 
-iree_lit_test_suite(
+iree_lit_test_suite(cfg = "//compiler:lit.cfg.py",
```
[the remark this answers is not in the document]
Comment (stellaraccident) [reply compiler/src/iree/compiler/Dialect/HAL/Target/CUDA/test/CMakeLists.txt:3]: Oops - one of my seds gone wrong. Thought I fixed all of those. Added a license header.

File: compiler/src/iree/compiler/README.md [hunk 3]
```diff
@@ -79,45 +78,3 @@ Read more:
 * https://google.github.io/styleguide/cppguide.html
 * https://llvm.org/docs/CodingStandards.html
 * https://clang.llvm.org/docs/ClangFormatStyleOptions.html
-
```
[the remark this answers is not in the document]
Comment (stellaraccident) [reply compiler/src/iree/compiler/README.md:82]: I don't love how the directory specific READMEs layer right now. Just going to bluntly fix and then reorganize later.

**Comment by GMNGeoffrey** [comment]
I believe this was part of #8955. Please remember to include the "why" in the PR description. Otherwise someone coming back from a month vacation might be really confused why every file in the project moved :stuck_out_tongue:

**Pull request merged**